lib/cluster/membership.c

```diff
@@ -84,15 +84,27 @@ static gboolean crm_autoreap  = TRUE;
 static void update_peer_uname(crm_node_t *node, const char *uname);
 static crm_node_t *find_known_node(const char *id, const char *uname);
 
-int
-crm_remote_peer_cache_size(void)
+/*!
+ * \internal
+ * \brief Get the number of Pacemaker Remote nodes that have been seen
+ *
+ * \return Number of cached Pacemaker Remote nodes
+ */
+guint
+pcmk__cluster_num_remote_nodes(void)
 {
     if (crm_remote_peer_cache == NULL) {
-        return 0;
+        return 0U;
     }
     return g_hash_table_size(crm_remote_peer_cache);
 }
 
+int
+crm_remote_peer_cache_size(void)
+{
+    return pcmk__cluster_num_remote_nodes();
```

Comment:
static analysis might complain about possible overflow, even though it's not realistic in practice

Reply:
You're right but this was existing behavior. `crm_remote_peer_cache_size()` returned `g_hash_table_size(crm_remote_peer_cache)` cast to an `int`, which might overflow

Reply:
I'm indifferent about changing it, just wasn't worried about it at the time

Reply:
easy enough to check against INT_MAX